### PR TITLE
Fix for windows path issue

### PR DESCRIPTION
**Pull request by jitu5**
## Description

Resolves  https://github.com/kedro-org/vscode-kedro/issues/102

This pull request includes several updates to the `bundled/tool/lsp_server.py` file, focusing on improving the way file URIs are resolved and refining the handling of pipeline files. The most important changes are as follows:

## Development notes

* Updated the `Location` object creation to use the `as_uri()` method for resolving file URIs in `_get_param_location`, `_query_catalog`, and `reference_location` methods.

* Refactored the `references` method to use `Path.read_text()` for reading pipeline files, which includes handling potential I/O and Unicode errors. 

## QA notes

Used [Parallels Desktop Pro](https://www.parallels.com/) to test on Windows.

## Gif (Windows)
windows-1

### CODE REVIEW

File: bundled/tool/lsp_server.py [hunk 1]
```diff
@@ -250,7 +250,7 @@ def _get_param_location(server: KedroLanguageServer, word: str) -> Optional[Loca
             continue
 
         location = Location(
-            uri=f"file://{parameters_file.resolve().as_posix()}",
+            uri=parameters_file.resolve().as_uri(),
```
Comment (noklam) [inline bundled/tool/lsp_server.py:253]:
```ipython
>>> path
PosixPath('C:\\Window\\sysm\\some_file.csv')
>>> path.resolve()
PosixPath('/Users/Nok_Lam_Chan/C:\\Window\\sysm\\some_file.csv')
>>> path.resolve().as_uri()
'file:///Users/Nok_Lam_Chan/C%3A%5CWindow%5Csysm%5Csome_file.csv'
```

Brilliant!

**Review by noklam — COMMENTED** [review]
thanks @jitu5, the fix looks good for me. One more thing to check: https://github.com/kedro-org/vscode-kedro/issues/53

Though the issue is about settings, but from the screenshot it looks like a Window issue with the select environment command. Have you check that as well?

File: bundled/tool/lsp_server.py (hunk 1 above)
Comment (rashidakanchwala) [reply bundled/tool/lsp_server.py:253]: same! TIL. nice :)

**Review by rashidakanchwala — APPROVED** [review]
I am not familiar with the code, and I don't have windows desktop to test it. But based on the issue around Windows Path, and looks like Nok tested it already, it looks good to me.

**Review by noklam — APPROVED** [review]
Approved in principle, let's wait a bit for the user to come back. Otherwise we can merge this in two days.

**Pull request merged**